docs/configuring.md

```diff
@@ -703,6 +703,47 @@ questions with default answers.
     copier -fd 'user_name=Manuel Calavera' copy template destination
     ```
 
+!!! info
+
+    The `--data` option is mutually exclusive with [`--data-file`](#data_file) option.
+
+### `data_file`
+
+-   Format: `str`
+-   CLI flags: `--data-file`
+-   Default value: N/A
+
+As an alternative you can also pass the path to a YAML file that contains your data.
+Passing a data file is only supported through the CLI.
+
+This cannot be defined in `copier.yml`, where its equivalent would be just normal
+questions with default answers.
+
+!!! example
+
+    Example CLI usage with a YAML file containing data:
+
+    ```yaml title="input.yml"
+    user_name: Manuel Calavera
+    age: 7
+    height: 1.83
+    ```
+
+    Passing a data file
+
+    ```shell
+    copier --data-file input.yml copy template destination
+    ```
+
+    is equivalent to passing its content as key-value pairs:
+
+    ```shell
+    copier -d 'user_name=Manuel Calavera' -d 'age=7' -d 'height=1.83' copy template destination
+
+!!! info
+
+    The `--data-file` option is mutually exclusive with [`--data`](#data) option.
```

Comment:
Juat a minor suggestion to retain the wording of the docs:

```suggestion
    The `--data-file` flag is mutually exclusive with the [`-d, --data`](#data) flags.
```

WDYT, @Ian2012?